LanguageFeatures/Patterns/type_inference_A29_t07.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// To type check a pattern p being matched against a value of type M:
+/// ...
+/// Map:
+/// i. Calculate the value's entry key type K and value type V, and key context
+///     C:
+///   a. If p has type arguments <K, V> for some K and V then use those, and C
+///     is K.
+///   b. Else if M implements Map<K, V> for some K and V then use those, and C
+///     is K.
+///   c. Else if M is dynamic then K and V are dynamic and C is _.
+///   d. Else K and V are Object? and C is _.
+/// ii. Type-check each key expression using C as the context type.
+/// iii. Type-check each value subpattern using V as the matched value type.
+/// vi. The required type of p is Map<K, V>.
+///
+/// @description Check that if `p` has no type arguments and `M` implements
+/// `Map<K, V>` then value's value type is `V` and key context is `K`
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/expect.dart";
+import "../../Utils/static_type_helper.dart";
+
+class C<T> {
+  const C();
```

Comment:
Github refuses to show me the comment from you on this line:

> Again, function below returns "no match". Wha't wrong?

Looks like a bug, the pattern key (line 36) should infer as `const C<String>()`.

Reply:
Ok